packages/lesswrong/components/hooks/useExternalScript.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from 'react';
+import { isClient } from '@/lib/executionEnvironment';
+
+type ExternalScriptState = {
+  loading: boolean
+  href: string
+  onReady: Array<() => void>
+};
+declare global {
+  interface Window {
+    externalScripts?: Array<ExternalScriptState>
+  }
+}
+
+/**
+ * Loads an external Javascript dependency, by injecting a <script> tag into
+ * the page header. Used for external client-side Javascript APIs such as
+ * Type3 Audio.
+ *
+ * Once a script is loaded this way, it stays loaded even if you navigate the
+ * tab and this hook is no longer present. This is because removing a script
+ * tag from the DOM will often not actually clean it up; repeatedly removing
+ * a script tag and then adding it back is likely to be a memory leak.
+ */
+export const useExternalScript = (href: string, scriptProps: Partial<Record<string,string>>): {
+  ready: boolean
+} => {
+  const [ready, setReady] = useState(() => {
+    if (isClient) {
+      const scriptState = window.externalScripts?.find(s => s.href === href)
+      if (!scriptState) return false;
+      return !scriptState.loading;
+    } else {
+      return false;
+    }
+  });
+
+  useEffect(() => {
+    const existingScriptState = window.externalScripts?.find(s => s.href === href)
+    
+    if (!existingScriptState) {
+      // If the script isn't loaded and its load hasn't been started it, start loading
+      if (!window.externalScripts) {
+        window.externalScripts = [];
+      }
+      const scriptState: ExternalScriptState = {loading: true,  href, onReady: []}
+      window.externalScripts.push(scriptState);
+      
+      const scriptTag = document.createElement("script");
+      scriptTag.async = true;
+      scriptTag.src = href;
+      for (let key of Object.keys(scriptProps)) {
+        (scriptTag as any)[key] = scriptProps[key];
+      }
+      scriptTag.onerror = () => {
+        // If the script fails to load, remove it from window.externalScripts
+        // so that if we navigate and return, it will try loading again.
+        if (window.externalScripts) {
+          window.externalScripts = window.externalScripts.filter(s => s.href !== href);
+        }
```

Comment:
Maybe `console.warn` something?

Reply:
I was assuming the default behavior would still hapen, and would involve logging something. I haven't tested injecting a fault here (I tried plug-pulling but that caused it to fail before it got far enough to see this particular error), but https://developer.mozilla.org/en-US/docs/Web/API/Window/error_event suggests that the default behavior will happen unless this `onerror` handler returns `true`.